generic/shared/src/main/scala/monocle/generic/Product.scala

```diff
@@ -1,11 +1,37 @@
 package monocle.generic
 
-import monocle.Iso
+import language.higherKinds
+import monocle.PTraversal
+import monocle.function.Each
+import monocle.{ Iso, Traversal }
 import monocle.generic.internal.TupleGeneric
+import scalaz.Applicative
+import scalaz.syntax.apply._
+import shapeless.{ ::, Generic, HList, HNil }
 
 object product extends ProductOptics
 
 trait ProductOptics {
   def productToTuple[S <: Product](implicit ev: TupleGeneric[S]): Iso[S, ev.Repr] =
     Iso[S, ev.Repr](s => ev.to(s))(t => ev.from(t))
-}
\ No newline at end of file
+
+  implicit def hNilEach[A] = new Each[HNil, A] {
+    def each: Traversal[HNil, A] = new PTraversal[HNil, HNil, A, A] {
```

Comment:
I think we could replace this by `Traversal.void[HNil, A]` as it is a noop traversal